src/app/routes/getInitialData/index.js

```diff
@@ -6,19 +6,48 @@ import onClient from '#lib/utilities/onClient';
 import getBaseUrl from './utils/getBaseUrl';
 import getPreprocessorRules from './utils/getPreprocessorRules';
 
-const logger = nodeLogger(__filename);
+export const logger = nodeLogger(__filename);
 const STATUS_CODE_OK = 200;
 const STATUS_CODE_BAD_GATEWAY = 502;
 const STATUS_CODE_NOT_FOUND = 404;
 const upstreamStatusCodesToPropagate = [STATUS_CODE_OK, STATUS_CODE_NOT_FOUND];
+const validRendererEnvironments = ['int', 'test', 'live'];
 
 const ampRegex = /(.amp)$/;
 
 const baseUrl = onClient()
   ? getBaseUrl(window.location.origin)
   : process.env.SIMORGH_BASE_URL;
 
-const getUrl = pathname => `${baseUrl}${pathname.replace(ampRegex, '')}.json`;
+// eslint-disable-next-line consistent-return
+export const getUrl = pathname => {
```

Comment:
There is a lot going on in this method, and I found it a little hard to follow on the first pass.

Do you think it would be worth trying a library for parsing query strings? This one looks reasonable and is very small (only 452b gzipped) https://www.npmjs.com/package/querystringify

This may or may not be helpful, however it would be worth investigating.

Reply:
Thanks for this - querystringify looks super easy to use, so I am going to have a look.